### PR TITLE
Dima/transform pose

### PR DESCRIPTION
**Pull request by dorezyuk**
Hey, 
I've added a small patch to transform the given poses into the global frame id of the global costmap - this is more or less an implicit requirement for many global planners.

### CODE REVIEW

**Comment by dorezyuk** [comment]
this kinetic stuff is really painful

File: mbf_costmap_nav/src/mbf_costmap_nav/costmap_planner_execution.cpp [hunk 1]
```diff
@@ -77,12 +75,26 @@ uint32_t CostmapPlannerExecution::makePlan(const geometry_msgs::PoseStamped &sta
                                            double &cost,
                                            std::string &message)
 {
+  // transform the input to the global frame of the costmap, since this is an
+  // "implicit" requirement for most planners
+  // note: costmap_2d::Costmap2DROS::getTransformTolerance might be a good idea,
+  // but it's not part of the class API in ros-kinetic
+  const ros::Duration pat(0.5);
```
Comment (corot) [inline mbf_costmap_nav/src/mbf_costmap_nav/costmap_planner_execution.cpp]: can we pass directly the values to transformPose?  or at least make frame a reference

**Review by corot — APPROVED** [review]
I have never faced the need of this, but sounds reasonable. I suppose most planners do the conversion themselves, e,g, [global_planner check the frames](https://github.com/ros-planning/navigation/blob/noetic-devel/global_planner/src/planner_core.cpp#L234)

The fact that u can provide start/goal poses in any reference frame should be documented somehow,,, maybe in GetPath.action?

File: mbf_costmap_nav/src/mbf_costmap_nav/costmap_planner_execution.cpp (hunk 1 above)
Comment (dorezyuk) [reply mbf_costmap_nav/src/mbf_costmap_nav/costmap_planner_execution.cpp]: the getGlobalFrameID returns by value for kinetic and melodic, so we cannot use a ref

Comment (corot) [reply mbf_costmap_nav/src/mbf_costmap_nav/costmap_planner_execution.cpp]: Ack;  can we rename pat as timeout?

**Review by spuetz — APPROVED** [review]

**Pull request merged**